lib/lz4.h

```diff
@@ -509,6 +509,17 @@ LZ4LIB_STATIC_API int LZ4_compress_fast_extState_fastReset (void* state, const c
  */
 LZ4LIB_STATIC_API void LZ4_attach_dictionary(LZ4_stream_t* workingStream, const LZ4_stream_t* dictionaryStream);
 
+/*! LZ4_getDictSize():
```

Comment:
Thanks for adding this documentation. It makes the intention clearer.
Yet, I had a hard time connecting the function's name with its intended objective:
> This can be used for adding data without compression to the LZ4 archive. 
> If linked blocked mode is used the memory of the dictionary is kept free.

I suspect that's because the documentation blends multiple layers of responsibilities in this paragraph.

At this place in the API, `LZ4_getDictSize()` seems to be _just_ about knowing the current dictionary size of the active `LZ4_stream_t*` state. And it's likely implied that this is not to be used in a concurrent access scenario.

That this function is employed in the context of `LZ4Frame` for a specific mode adding uncompressed data can be interesting information, but it does not define what this function is doing. The size information it provides could be employed for any other usage, so it matters that it's cleanly defined.

This leads me to a few simple questions : 
- what is `@dictSize` argument for ?
  All it does is cap the reported `dictSize`, without changing anything to underlying situation ?
  If the point of this function is to return the dictionary size, maybe it should do just that ?
  And if there is a reason to cap the value at the calling site, maybe this should be done at the calling site ?
- Getter general to not mutate the state they are looking into. If that's the case here too, the state could be `const LZ4_stream_t*` instead, which makes it clear that this function has no side effect.

Reply:
This code just has been moved out of the `LZ4_saveDict` function (where it's also used now). 

I passed `@dictSize` to be consistent with the previous implementation. I'm fine with removing the dict size parameter and capping it to 64 KB locally in this function. But than we would either have to change the signature of `LZ4_saveDict` to remove the `dictSize` parameter or restore the old code of `LZ4_saveDict` to calculate the dict size there, which would lead to duplicated code. 

As for you second point: I changed the parameter as well as `dict` `const` to make clear that these are not modified. 

As I wrote in the other thread already all this only has been added so do not modify the dictionary when adding uncompressed data. 
If you think modifying the dictionary is okay when adding uncompressed data, I'd remove the `if` from https://github.com/lz4/lz4/pull/1094/files#diff-16e71ed5519d7ce479c3a3c3158b3e5b121fd300b78497bb477a6695b6d08b50R969 and restore the old way the dict was calculated here. 

In case we keep the `get_dictSize` function I'll update the documentation again to again make it a bit clearer what the intent of this function is.